packages/@aws-cdk/aws-s3/test/auto-delete-objects-handler.test.ts

```diff
@@ -55,6 +95,45 @@ test('does nothing on update event', async () => {
   expect(mockS3Client.deleteObjects).toHaveBeenCalledTimes(0);
 });
 
+test('deletes all objects when the name changes on update event', async () => {
```

Comment:
add a test case for when OldProperties and NewProperties exist, the bucket names are same but something else is different.